File: common/dto/document.go
package dto

type CreateDocument struct {
	Title  string `json:"title"`
	Text   string `json:"text"`
	Status string `json:"status" validate:"required" enum:"draft,completed,doing,rejected"`
}

type UpdateDocument struct {
	Title  string `json:"title"`
	Text   string `json:"text"`
	Status string `json:"status" enum:"draft,completed,doing,rejected"`
}

File: web/src/types/document.ts
export type DocumentStatus = "completed" | "draft" | "rejected" | "review";

export type CreateDocument = {
    title: string;
    text: string;
    status: DocumentStatus;
};

export type Document = {
    id: number;
    title: string;
    text: string;
    status: DocumentStatus;
    createdAt: string;
    updatedAt: string;
};
